src/Aquifer.Common/Jobs/QueueClientFactory.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System.Collections.Concurrent;
+using Aquifer.Common.Services;
+using Azure.Storage.Queues;
+
+namespace Aquifer.Common.Jobs;
+
+public interface IQueueClientFactory
+{
+    public Task<QueueClient> GetQueueClientAsync(string queueName, CancellationToken ct);
+}
+
+public sealed class QueueClientFactory(QueueConfigurationOptions _queueConfigurationOptions, IAzureClientService _azureClientService)
```

Comment:
The goal of this class is to create a factory that allows for lazily ensuring that a queue exists upon accessing a queue client for publishing purposes.  This will only happen once per queue per app service lifetime and only if we try to publish to that queue.  Previous code did this web request to ensure the queue exists in the class constructor which isn't a best practice.